core/shared/src/main/scala/zio/prelude/Equal.scala

```diff
@@ -863,7 +863,7 @@ object Equal extends EqualVersionSpecific {
    * Since there is only one `Unit` value all equality comparisons will always be true.
    */
   implicit val UnitHashOrd: Hash[Unit] with Ord[Unit] =
-    HashOrd.make(_.hashCode, (_, _) => Ordering.Equals, (_, _) => true)
+    HashOrd.make(_ => 0, (_, _) => Ordering.Equals, (_, _) => true)
```

Comment:
`().hashCode` generates a warning: "dubious usage of method hashCode with unit value"
I'm not sure what it means.
According to https://www.scala-lang.org/api/2.13.11/scala/Unit.html,
> There is only one value of type Unit, (), and it is not represented by any object in the underlying runtime system
Which would mean that calling `.hasCode` on `()` doesn't have much sens as there's not instance of `Unit` at runtime

I'm not sure why this code was compiling and isn't anymore

Why did I decided to replace it with `_ => 0`?
Well, I've found this Akka piece of code: https://github.com/thadeurc/sagat/blob/master/code/akka-1.0-rc3-with-amqp/akka/akka-actor/src/main/scala/akka/util/HashCode.scala

So, in a worksheet, I tried to execute this code:
```scala
import java.lang.reflect.{Array => JArray}
import java.lang.{Float => JFloat, Double => JDouble}

object HashCode {
  val SEED = 23

  def hash(seed: Int, any: Any): Int = any match {
    case value: Boolean => hash(seed, value)
    case value: Char => hash(seed, value)
    case value: Short => hash(seed, value)
    case value: Int => hash(seed, value)
    case value: Long => hash(seed, value)
    case value: Float => hash(seed, value)
    case value: Double => hash(seed, value)
    case value: Byte => hash(seed, value)
    case value: AnyRef =>
      var result = seed
      if (value eq null) result = hash(result, 0)
      else if (!isArray(value)) result = hash(result, value.hashCode())
      else for (id <- 0 until JArray.getLength(value)) result = hash(result, JArray.get(value, id)) // is an array
      result
  }
  def hash(seed: Int, value: Boolean): Int = firstTerm(seed) + (if (value) 1 else 0)
  def hash(seed: Int, value: Char): Int = firstTerm(seed) + value.asInstanceOf[Int]
  def hash(seed: Int, value: Int): Int = firstTerm(seed) + value
  def hash(seed: Int, value: Long): Int = firstTerm(seed)  + (value ^ (value >>> 32) ).asInstanceOf[Int]
  def hash(seed: Int, value: Float): Int = hash(seed, JFloat.floatToIntBits(value))
  def hash(seed: Int, value: Double): Int = hash(seed, JDouble.doubleToLongBits(value))

  private def firstTerm(seed: Int): Int = PRIME * seed
  private def isArray(anyRef: AnyRef): Boolean = anyRef.getClass.isArray
  private val PRIME = 37
}

HashCode.hash(HashCode.SEED, "Hello") // returns `69610501`
HashCode.hash(HashCode.SEED, ())      // returns `851`
().hashCode()                         // returns `0`
```

So I thought `0` would be the correct hashcode value for `Unit`

Reply:
We definitely wan to behave like `hashCode`.